crates/rpc/rpc/src/eth/revm_utils.rs

```diff
@@ -86,7 +94,15 @@ impl FillableTransaction for TransactionSigned {
     fn try_fill_tx_env(&self, tx_env: &mut TxEnv) -> EthResult<()> {
         let signer =
             self.recover_signer().ok_or_else(|| EthApiError::InvalidTransactionSignature)?;
+        #[cfg(not(feature = "optimism"))]
         fill_tx_env(tx_env, self, signer);
+
+        #[cfg(feature = "optimism")]
+        {
+            let mut envelope_buf = Vec::with_capacity(self.length_without_header());
+            self.encode_enveloped(&mut envelope_buf);
+            fill_tx_env(tx_env, self, signer, envelope_buf.into());
+        }
```

Comment:
ugh i really don't like how we're forced to re-serialize, no way to avoid this? assumign you guys have thought about this